sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1069,11 +1110,19 @@ class Analyzer(
         case a @ Aggregate(grouping, expressions, child) =>
           failOnOuterReference(a)
           val referencesToAdd = missingReferences(a)
+          if (foundNonEqualCorrelatedPred) {
+            failOnNonEqualCorrelatedPredicate(a)
+          }
           if (referencesToAdd.nonEmpty) {
             Aggregate(grouping ++ referencesToAdd, expressions ++ referencesToAdd, child)
           } else {
             a
           }
+        case w : Window =>
+          if (foundNonEqualCorrelatedPred) {
+            failOnNonEqualCorrelatedPredicate(w)
+          }
+          w
```

Comment:
We could do that if all the correlated predicates are equality predicates. It is as if the computation of the subquery is bound by each value from the outer reference and therefore can be bound in the PARTITION BY clause of a Window function.

I guess we want to implement this transformation outside of this PR, preferrably in Optimizer phase.